src/code/z_actor.c

```diff
@@ -236,77 +236,89 @@ void Actor_ProjectPos(PlayState* play, Vec3f* src, Vec3f* xyzDest, f32* cappedIn
-    TargetContextEntry* entry;
-    NaviColor* naviColor;
+void Target_InitReticle(TargetContext* targetCtx, s32 actorCategory, PlayState* play) {
+    LockOnReticle* reticle;
+    TargetColor* reticleColor = &sTargetColorList[actorCategory];
     s32 i;
 
     Math_Vec3f_Copy(&targetCtx->targetCenterPos, &play->view.eye);
-    targetCtx->unk_44 = 500.0f;
-    targetCtx->unk_48 = 0x100;
 
-    naviColor = &sNaviColorList[actorCategory];
+    targetCtx->reticleRadius = 500.0f;
+    targetCtx->unk_48 = 256;
+
+    reticle = &targetCtx->lockOnReticles[0];
+
+    for (i = 0; i < ARRAY_COUNT(targetCtx->lockOnReticles); i++, reticle++) {
+        Target_SetReticlePos(targetCtx, i, 0.0f, 0.0f, 0.0f);
 
-    entry = &targetCtx->arr_50[0];
-    for (i = 0; i < ARRAY_COUNT(targetCtx->arr_50); i++) {
-        func_8002BE64(targetCtx, i, 0.0f, 0.0f, 0.0f);
-        entry->color.r = naviColor->inner.r;
-        entry->color.g = naviColor->inner.g;
-        entry->color.b = naviColor->inner.b;
-        entry++;
+        reticle->color.r = reticleColor->inner.r;
+        reticle->color.g = reticleColor->inner.g;
+        reticle->color.b = reticleColor->inner.b;
     }
 }
 
-void Actor_SetNaviToActor(TargetContext* targetCtx, Actor* actor, s32 actorCategory, PlayState* play) {
-    NaviColor* naviColor = &sNaviColorList[actorCategory];
+void Target_SetNaviState(TargetContext* targetCtx, Actor* actor, s32 actorCategory, PlayState* play) {
+    TargetColor* targetColor = &sTargetColorList[actorCategory];
+
     targetCtx->naviRefPos.x = actor->focus.pos.x;
     targetCtx->naviRefPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
     targetCtx->naviRefPos.z = actor->focus.pos.z;
-    targetCtx->naviInner.r = naviColor->inner.r;
-    targetCtx->naviInner.g = naviColor->inner.g;
-    targetCtx->naviInner.b = naviColor->inner.b;
-    targetCtx->naviInner.a = naviColor->inner.a;
-    targetCtx->naviOuter.r = naviColor->outer.r;
-    targetCtx->naviOuter.g = naviColor->outer.g;
-    targetCtx->naviOuter.b = naviColor->outer.b;
-    targetCtx->naviOuter.a = naviColor->outer.a;
-}
 
-void func_8002C0C0(TargetContext* targetCtx, Actor* actor, PlayState* play) {
-    targetCtx->arrowPointedActor = targetCtx->targetedActor = targetCtx->unk_8C = targetCtx->bgmEnemy = NULL;
+    targetCtx->naviInner.r = targetColor->inner.r;
+    targetCtx->naviInner.g = targetColor->inner.g;
+    targetCtx->naviInner.b = targetColor->inner.b;
+    targetCtx->naviInner.a = targetColor->inner.a;
 
+    targetCtx->naviOuter.r = targetColor->outer.r;
+    targetCtx->naviOuter.g = targetColor->outer.g;
+    targetCtx->naviOuter.b = targetColor->outer.b;
+    targetCtx->naviOuter.a = targetColor->outer.a;
+}
+
+void Target_Init(TargetContext* targetCtx, Actor* actor, PlayState* play) {
+    targetCtx->arrowPointedActor = NULL;
+    targetCtx->targetedActor = NULL;
+    targetCtx->unk_40 = 0.0f;
+    targetCtx->unk_8C = NULL;
+    targetCtx->bgmEnemy = NULL;
```

Comment:
going to guess this is why retail isn't matching in your PR

Reply:
yeah guess so. chained assign really sucks here so itd be nice if it was possible without it, but if not then oh well

Reply:
you can match like
```
    // clang-format on
    targetCtx->bgmEnemy = NULL; \
    targetCtx->unk_8C = NULL; \
    targetCtx->targetedActor = NULL; \
    targetCtx->arrowPointedActor = NULL;
    // clang-format off
```
too, maybe that's the less bad option?

Reply:
oh actually chained assign isnt terrible in this case. For some reason I was under the impression that `unk_40` had to be in the chain, and setting NULL to that would be silly. But I think I just hallucinated that this was the case